src/libutil/logging.cc

```diff
@@ -344,15 +344,16 @@ std::unique_ptr<Logger> makeJSONLogger(const std::filesystem::path & path, bool
 void applyJSONLogger()
 {
     if (!loggerSettings.jsonLogPath.get().empty()) {
-        std::vector<std::unique_ptr<Logger>> loggers;
         try {
-            loggers.push_back(std::move(logger));
+            std::vector<std::unique_ptr<Logger>> loggers;
             loggers.push_back(makeJSONLogger(std::filesystem::path(loggerSettings.jsonLogPath.get()), false));
-            logger = makeTeeLogger(std::move(loggers));
+            // Note: this had better not throw, otherwise `logger` is
+            // left unset.
```

Comment:
Would it be worth adding an assert (or something like it) that the `logger` is set at the end of this function, so we can know if this happened?